pal/inc/CCSpinLock.hpp

```diff
@@ -0,0 +1,34 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#ifndef CC_PAL_INC_CCSPINLOCK_H
+#define CC_PAL_INC_CCSPINLOCK_H
+
+class CCSpinLock
+{
+    unsigned int  enterCount;
+    bool          shouldTrackThreadId;
+    char          lock;
+    size_t        threadId;
+public:
+    void Reset(bool trackThread = false)
```

Comment:
>void Reset(bool trackThread = false) [](start = 4, length = 36)

Is this "Reset" API really needed? Seems it adds risk. Assume you need this API... do you need to allow changing "shouldTrackThreadId"? If not, seems there be an opportunity to make that a const field or a template argument. Those are 2 styles of SpinLock.